tests/ProjectWideWrapperTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ObjectEncryptor\Tests;
+
+use Keboola\ObjectEncryptor\Exception\ApplicationException;
+use Keboola\ObjectEncryptor\Exception\UserException;
+use Keboola\ObjectEncryptor\Wrapper\ProjectAKVWrapper;
+use Keboola\ObjectEncryptor\Wrapper\ProjectKMSWrapper;
+use Keboola\ObjectEncryptor\Wrapper\ProjectWideAKVWrapper;
+use Keboola\ObjectEncryptor\Wrapper\ProjectWideKMSWrapper;
+use PHPUnit\Framework\TestCase;
+
+class ProjectWideWrapperTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        putenv('AWS_ACCESS_KEY_ID=' . getenv('TEST_AWS_ACCESS_KEY_ID'));
+        putenv('AWS_SECRET_ACCESS_KEY='. getenv('TEST_AWS_SECRET_ACCESS_KEY'));
+        putenv('AZURE_TENANT_ID=' . getenv('TEST_TENANT_ID'));
+        putenv('AZURE_CLIENT_ID=' . getenv('TEST_CLIENT_ID'));
+        putenv('AZURE_CLIENT_SECRET=' . getenv('TEST_CLIENT_SECRET'));
+    }
+
+    /**
+     * @return ProjectWideAKVWrapper[][]|ProjectWideKMSWrapper[][]
+     */
+    public function wrapperProvider(): array
+    {
+        $projectWrapperKMS = new ProjectWideKMSWrapper();
+        $projectWrapperKMS->setKMSRegion((string) getenv('TEST_AWS_REGION'));
+        $projectWrapperKMS->setKMSKeyId((string) getenv('TEST_AWS_KMS_KEY_ID'));
+
+        $projectWrapperAKV = new ProjectWideAKVWrapper();
+        $projectWrapperAKV->setKeyVaultUrl((string) getenv('TEST_KEY_VAULT_URL'));
+
+        return [
+            'KMS' => [
+                $projectWrapperKMS,
+            ],
+            'AKV' => [
+                $projectWrapperAKV,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider wrapperProvider
+     * @param ProjectWideKMSWrapper|ProjectWideAKVWrapper $wrapper
+     */
+    public function testEncrypt($wrapper): void
+    {
+        $secret = 'mySecretValue';
+        $wrapper->setStackId('my-stack');
+        $wrapper->setProjectId('my-project');
+        $encrypted = $wrapper->encrypt($secret);
+        self::assertNotEquals($secret, $encrypted);
+        self::assertEquals($secret, $wrapper->decrypt($encrypted));
+
+        $wrapper->setStackId('my-stack');
+        $wrapper->setProjectId('my-project');
+        self::assertEquals($secret, $wrapper->decrypt($encrypted));
```

Comment:
Tohle udela co? Prijde mi to uplne stejne jako assert vys

Reply:
Jo to je tam zbytečný